crates/router/src/core/admin.rs

```diff
@@ -219,6 +224,14 @@ pub async fn merchant_account_update(
         locker_id: req
             .locker_id
             .or_else(|| merchant_account.locker_id.to_owned()),
+        metadata: if req.metadata.is_some() {
+            Some(
+                utils::Encode::<api::MerchantDetails>::encode_to_value(&req.metadata)
+                    .change_context(errors::ApiErrorResponse::InternalServerError)?,
+            )
+        } else {
+            merchant_account.metadata.to_owned()
+        },
```

Comment:
Try using
```rust
req.metadata
	.as_ref()
	.map(...
	.change_context(...
	.or_else(|| merchant_account.metadata.to_owned()),
```

Reply:
If you can, also use a similar pattern to handle the `req.custom_routing_rules` above